### ISSUE TITLE
Use Libloading

### ISSUE THREAD

**Issue by ghost** [YeyaSwizaw/rust-snek#1]
Not a bug but a suggestion, [libloading](https://github.com/nagisa/rust_libloading/) will give you windows support that you don't have to maintain.

Of course it has its own way of dealing with symbols being unloaded that is duplicated in your project already.

**Comment by samuelsleight** [comment]
Yes, this library could have just been a macro around an existing project like libloading, however I wanted to implement the backend myself (the unix one, at least - I had difficulty getting the windows support working) for educational purposes. I may switch to using it or similar in the future though.
